test/jdk/java/net/httpclient/HttpResponseLimitingTest.java

```diff
@@ -0,0 +1,273 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8328919
+ * @summary tests `limiting()` in `HttpResponse.Body{Handlers,Subscribers}`
+ * @library /test/lib
+ *          /test/jdk/java/net/httpclient/lib
+ * @build jdk.test.lib.net.SimpleSSLContext
+ *        jdk.httpclient.test.lib.common.HttpServerAdapters
+ * @run junit HttpResponseLimitingTest
+ */
+
+import jdk.httpclient.test.lib.common.HttpServerAdapters;
+import jdk.test.lib.net.SimpleSSLContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.net.http.HttpResponse.BodySubscriber;
+import java.net.http.HttpResponse.BodySubscribers;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Flow.Subscription;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.copyOfRange;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class HttpResponseLimitingTest {
+
+    private static final byte[] RESPONSE_BODY = "random non-empty body".getBytes(UTF_8);
+
+    @ParameterizedTest
+    @MethodSource("sufficientCapacities")
+    void testSuccessOnSufficientCapacity(HttpClient.Version version, boolean secure, long sufficientCapacity) throws Exception {
+        HttpResponse<byte[]> response = requestBytes(version, secure, sufficientCapacity);
+        assertArrayEquals(RESPONSE_BODY, response.body());
+    }
+
+    static Arguments[] sufficientCapacities() {
+        return capacityArgs(Long.MAX_VALUE, RESPONSE_BODY.length);
+    }
+
+    @ParameterizedTest
+    @MethodSource("insufficientCapacities")
+    void testFailureOnInsufficientCapacity(HttpClient.Version version, boolean secure, long insufficientCapacity) {
+        assertThrows(
+                IOException.class,
+                () -> requestBytes(version, secure, insufficientCapacity),
+                "body exceeds capacity: " + RESPONSE_BODY.length);
```

Comment:
The 3rd parameter here is the message representing the failure of the assert. In its current form if this test fails (i.e. the IOException isn't thrown), then the failure message will say "body exceeds capacity: X" which I think is a bit misleading since it gives an impression that the response body has exceeded capacity.

I think changing that message to something like `"No IOException thrown for capacity: " + insufficientCapacity` might be a bit more clear.

Reply:
Fixed in fc4d674496f26501f8203cd2f77c3d42f98bfdf6.